src/button_handler.cpp

```diff
@@ -179,6 +218,14 @@ void Handler::longPowerPressed(sdbusplus::message::message& msg)
 {
     try
     {
+        if (!isBmcReady())
```

Comment:
For the long press case, I think this check should be after we check for the poweredOn below, and only if we are not powered on.

The reasoning being that the BMC state should not matter if we are powered ON.

However, the current implementation seems to ignore a long press completely if the system is not powered ON. Since the long press is only used to turn the power OFF, I think we can omit the BMC state check altogether.

Reply:
Yes, longPowerPressed ignored. The current implementation is now a bit similar to FSP design - handling accidental/DPO/FPO. We can ignore the check as well.